editor/src/components/navigator/left-pane/pages-pane.tsx

```diff
@@ -524,30 +562,231 @@ export const AddPageContextMenu = React.memo(
 function useScrollToNavigateToRoute(
   ref: React.MutableRefObject<HTMLDivElement | null>,
   path: string,
-  navigateTo: string | null,
+  navigateTo: NavigateTo | null,
 ) {
   React.useEffect(() => {
-    if (navigateTo === path) {
+    if (navigateTo?.resolvedPath === path) {
       ref.current?.scrollIntoView()
     }
   }, [navigateTo, ref, path])
 }
 
 function useNavigateToRouteWhenAvailable(
   remixRoutes: RouteMatch[],
-  navigateTo: string | null,
+  navigateTo: NavigateTo | null,
   onNavigate: () => void,
 ) {
   const [navigationControls] = useAtom(RemixNavigationAtom)
   const [activeRemixScene] = useAtom(ActiveRemixSceneAtom)
 
+  const scenesToNavigate: string[] = React.useMemo(() => {
+    if (navigateTo == null) {
+      return []
+    }
+    switch (navigateTo.mode) {
+      case 'all-scenes':
+        return Object.keys(navigationControls)
+      case 'only-active-scene':
+        return [EP.toString(activeRemixScene)]
+      default:
+        assertNever(navigateTo.mode)
+    }
+  }, [navigationControls, activeRemixScene, navigateTo])
+
   React.useEffect(() => {
     if (navigateTo == null) {
       return
     }
-    if (remixRoutes.some((r) => r.resolvedPath === navigateTo)) {
-      void navigationControls[EP.toString(activeRemixScene)]?.navigate(navigateTo)
+
+    // if the target is a specific route, go to it
+    const navigateToMatchesRealRoute = remixRoutes.find((r) => {
+      return r.resolvedPath === navigateTo.resolvedPath
+    })?.matchesRealRoute
+
+    // otherwise if the target is not a specific route, go to the first valid match for its prefix
+    const navigationTarget = navigateToMatchesRealRoute
+      ? navigateTo.resolvedPath
+      : remixRoutes.find((r) => r.matchesRealRoute && r.path.startsWith(navigateTo.routePath))
+          ?.resolvedPath ?? null
+
+    if (navigationTarget != null) {
+      scenesToNavigate.forEach((path) => {
+        void navigationControls[path]?.navigate(navigationTarget)
+      })
       onNavigate()
     }
-  }, [navigateTo, remixRoutes, navigationControls, activeRemixScene, onNavigate])
+  }, [navigateTo, remixRoutes, navigationControls, activeRemixScene, onNavigate, scenesToNavigate])
 }
+
+function replacementTokensMatch(oldTokens: string[], newTokens: string[]): boolean {
+  if (oldTokens.length !== newTokens.length) {
```

Comment:
I think you can use `arrayEqualsByReference` for this.

Reply:
oh sweet

Reply:
[`91b96aa` (#5265)](https://github.com/concrete-utopia/utopia/pull/5265/commits/91b96aa047c66431f9fcce413b180aaa2e9cfd54)